dvc/remote/local.py

```diff
@@ -151,6 +151,9 @@ def remove(self, path_info):
             raise NotImplementedError
 
         if self.exists(path_info):
+            checksum = self.get_checksum(path_info)
```

Comment:
@sharidas, looks like on some tests, the state does not get load and `get_checksum` fails?

@pared, I wonder if this could be handled here: https://github.com/iterative/dvc/blob/85c35f894cf12ddf26eb63abbcd5660dcea4e909/dvc/remote/base.py#L699-L700

Reply:
@skshetry Not sure. I was looking into one of the test, basically looking into the stack trace and found https://github.com/iterative/dvc/blob/master/tests/func/test_checkout.py#L320. And this test does not parse the gc code if I noticed the flow correctly.

Reply:
The updated `.remove()` has `.get_checksum()` which gets checksum from `.dvc/state` AFAIK, but the test does not load the state and hence the exception. You can load the state via `with self.dvc.state:` though. But, I'd try moving the logic after this [line](https://github.com/iterative/dvc/blob/85c35f894cf12ddf26eb63abbcd5660dcea4e909/dvc/remote/base.py#L699) (though, a check if it's remote is required), but @iterative/dvc developers might have a different suggestion.

Reply:
@skshetry Is right and that is also how we've described the solution in https://github.com/iterative/dvc/issues/2946#issuecomment-569779763 . This PR is trying to make a change in a generic method that doesn't really need to know anything about checksums and is often used in other applications where it is not expected to have such a behaviour.

Reply:
@efiop Thanks for the feedback. I was trying to make the solution close to the remote local. Now I have updated the pr again by making change in base.py. Initially I made the change in the base.py as suggested in the issue. Later I thought it would be better to make the solution close to the local.

Reply:
@sharidas Makes sense, I guess my comment about "local" was confusing, sorry about that :slightly_smiling_face: